src/qiskit_qasm3_import/expression.py

```diff
@@ -62,43 +76,60 @@ class ValueResolver(QASMVisitor):
     def __init__(self, symbols: Mapping[str, Symbol]):
         self.symbols = symbols
 
-    def resolve(self, node: ast.Expression) -> Tuple[Any, types.Type]:
+    def resolve(self, node: ast.Expression, context=None) -> Tuple[Any, types.Type]:
         """The entry point to the resolver, resolving the AST node into a 2-tuple of a relevant
         Qiskit type, and the :class:`.Type` that it is an instance of."""
-        return self.visit(node)
+        if context is None:
+            return self.visit(node)
+        return self.visit(node, context)
 
     def visit(self, node: ast.QASMNode, context: None = None) -> Tuple[Any, types.Type]:
-        value, type = super().visit(node, context)
+        if context is None:
+            value, type = super().visit(node)
+        else:
+            value, type = super().visit(node, context)
```

Comment:
Whatever we do with `None` handling, I don't think there's any need to do this - the super `visit` method will do it.